files-community/Files
# Code Quality: Introduced RichCommand parameter to IExtendedAction

## Summary

Executing RichCommand from a layout area never requires parameters. However, when calling from the sidebar or the widget, there is no selected item, so you will need to pass a parameter. To meet this requirements, we introduced parameters to certain commands.

Now, you can intoruce RichCommand even for sidebar items, widget items and so on.

## Task Checklist

- [x] Introduced `IExtendedAction` that have parameter property
- [x] Introduced `EjectDrive` RichCommand
- [x] Introduced `IExtendedAction` into some Actions
- [x] Introduced `DriveItemContextFlyoutFactory`
- [x] Renamed `ContextMenuFlyoutItemViewModel` with `CustomMenuFlyoutItem`
- [x] Renamed `ContextMenuFlyoutItemViewModelBuilder` with `CustomMenuFlyoutItemBuilder`
- [x] Renamed `ContextFlyoutItemHelper` with `ContentContextFlyoutFactory`
- [x] Renamed `ItemModelListToContextFlyoutHelper` with `ItemToCommandBarConversionHelper`
- [x] Changed property `IsHidden` to `IsVisible` in `CustomMenuFlyoutItem`
- [x] Renamed property `ShowItem` with `IsAvailable` in `CustomMenuFlyoutItem`
- [x] Intoroduced a builder for RichCommand inside `CustomMenuFlyoutItem` to make `CustomMenuFlyoutItemBuilder` deprecated
- [x] Renamed `ContextFlyoutItemHelper` with `ContentContextFlyoutFactory`
- [x] Renamed `ItemModelListToContextFlyoutHelper` with `ItemToCommandBarConversionHelper`
- [x] Deprecated `MenuFlyoutHelper` class

## Known Issues

_N/A_

## PR Checklist

- [x] **Closes**: Related #13892

## Steps To Validate Changes

1. Launch the app
2. Go to Drive Widgets
3. Right tap a drive card
4. Click all items and check if it work correctly
   - Open in new tab
   - Open in new window
   - Open in side pane
   - Eject
   - Format
   - Pin/unpin
   - Open properties window

## Screenshots

Snapshot what you'd be most likely to see:
![image](https://github.com/files-community/Files/assets/62196528/0856dfbb-a8fd-4d95-8024-7e8927ef2f80)

[^1]: If the request hasn't been agreed by the team, this work might be rejected. Make sure that you get approval from the team before opening any PR related the request.
[^2]: If you removed any en-US string resources, make sure that there are no references of those resources. When you add a new en-US string resources, do not make any changes on other languages' resources; [Crowdin](https://crowdin.com/project/files-app) will do that, instead.
